src/jit/emitxarch.cpp

```diff
@@ -5519,11 +5519,28 @@ static bool isSseShift(instruction ins)
     }
 }
 
+static bool isSseShuffle(instruction ins)
+{
+    switch (ins)
+    {
+        case INS_pshufb:
+        case INS_pshufd:
+        case INS_pshufhw:
+        case INS_pshuflw:
+        case INS_shufpd:
+        case INS_shufps:
+            return true;
+        default:
+            return false;
+    }
+}
+
 void emitter::emitIns_SIMD_R_R_I(instruction ins, emitAttr attr, regNumber reg, regNumber reg1, int ival)
 {
     // TODO-XARCH refactoring emitIns_R_R_I to handle SSE2/AVX2 shift as well as emitIns_R_I
-    bool isShift = isSseShift(ins);
-    if (UseVEXEncoding() && !isShift)
+    bool isShift   = isSseShift(ins);
+    bool isShuffle = isSseShuffle(ins);
+    if ((UseVEXEncoding() && !isShift) || isShuffle)
```

Comment:
I believe that you do not need `isSseShuffle`, these shuffle instructions should be in `IsDstDstSrcAVXInstruction`.

Reply:
`isSseShuffle` is used only for SSE2 legacy encoding codepath and this path is not hit when emitting `vpshufd`. 

`IsDstDstSrcAVXInstruction` has no impact on SSE2 legacy codegen. After adding `pshufd` to `IsDstDstSrcAVXInstruction` codegen for VEX encoded instruction for optimized ro test was emitting invalid instruction - at least test was failing with native invalid instruction exception.

JIT disasm was identical for both VEX cases with `IsDstDstSrcAVXInstruction` and without `IsDstDstSrcAVXInstruction`, so I am really puzzled by this.
```
C4E17970C0FF         vpshufd  xmm0, xmm0, 255 - no DstDstSrc
C4E17970C0FF         vpshufd  xmm0, xmm0, 255 - with DstDstSrc
```
[ShuffleRoDisasm.txt](https://github.com/dotnet/coreclr/files/1773109/ShuffleRoDisasm.txt)
[ShuffleRoNoDstDstSrcDisasm.txt](https://github.com/dotnet/coreclr/files/1773110/ShuffleRoNoDstDstSrcDisasm.txt)

Reply:
Ah, I see. These SSE2 shuffle instructions have the special encoding.
Okay, the current code looks correct, sorry.

Reply:
> vpshufhw and vpshuflw are emitted correctly when added to IsDstDstSrcAVXInstruction

No, they should not be in `IsDstDstSrcAVXInstruction`.

I looked the manual again. These intrinsics have 4 instructions:
`PSHFD`
`PSHUFHW`
`PSHUFLW`
`SHUFPD`

`PSHFD`, `PSHUFHW`, and `PSHUFLW` should be specially treated here because they have the same register-form in SSE and VEX encoding.
`SHUFPD` should be in  `IsDstDstSrcAVXInstruction` because it has VEX.NDS encoding.

Reply:
OK - I will remove all of them.

Reply:
Done - left only `SHUFPD`